### PR TITLE
Add Prometheus support for monitoring MinIO ports

### PR DESCRIPTION
**Pull request by nitisht**
To Test locally:

- Deploy M3 as usual, add a tenant.
- `kubectl get pods`
- Get the Pod name of Prometheus.
- Use `kubectl port-forward <prometheus-pod-name> 9090:9090`
- Open up localhost:9090, you should be able to see Prometheus dashboard with MinIO tenant containers as targets.
- To test auto discovery, add another tenant.
- Prometheus dashboard should automatically update with new MinIO tenant containers

### CODE REVIEW

File: k8s/create-kind.sh [hunk 1]
```diff
@@ -2,16 +2,18 @@
 
 echo "Provisioning Kind"
 kind create cluster --name m3cluster --config kind-cluster.yaml
+export KUBECONFIG="$(kind get kubeconfig-path --name="m3cluster")"
```
Comment (dvaldivia) [inline k8s/create-kind.sh]: this is no longer needed with the latest kind `0.6.1`

File: k8s/create-kind.sh [hunk 2]
```diff
@@ -2,16 +2,18 @@
...
 echo "creating service account"
 kubectl create serviceaccount dashboard -n default
 kubectl create clusterrolebinding dashboard-admin -n default --clusterrole=cluster-admin --serviceaccount=default:dashboard
-# pre-load MinIO, postgres, etcd, etcd-operator to speed up setup
+# pre-load MinIO, postgres, etcd, etcd-operator, Prometheus to speed up setup
 docker pull minio/minio:RELEASE.2019-12-17T23-16-33Z
 docker pull postgres:12
 docker pull quay.io/coreos/etcd-operator:v0.9.4
 docker pull quay.io/coreos/etcd:v3.4.0
+docker pull quay.io/prometheus/prometheus:v2.14.0
```
Comment (dvaldivia) [inline k8s/create-kind.sh:15]: you should also load the prometheus image into kind with `kind load docker-image quay.io/prometheus/prometheus:v2.14.0`

File: k8s/create-kind.sh [hunk 3]
```diff
@@ -22,6 +24,9 @@ kubectl apply -f deployments/m3-portal-backend-deployment.yaml
 kubectl apply -f deployments/m3-portal-frontend-deployment.yaml
 kubectl apply -f deployments/portal-proxy-deployment.yaml
 
+# create Prometheus configMap, this will be used by Prometheus Kubernetes Deployment
+kubectl apply -f deployments/prometheus-configmap.yaml
```
Comment (dvaldivia) [inline k8s/create-kind.sh]: this file is only used by developers, sounds like this config map is something that will be needed by production deployments, maybe embed this config map into `m3-deployment.yaml.example` to make sure people configure it properly

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: k8s/create-kind.sh (hunk 2 above)
Comment (nitisht) [reply k8s/create-kind.sh:15]: done

File: k8s/create-kind.sh (hunk 3 above)
Comment (nitisht) [reply k8s/create-kind.sh]: makes sense, done!

**Review by cesnietor — DISMISSED** [review]

**Review by dvaldivia — DISMISSED** [review]

**Comment by cesnietor** [comment]
@nitisht  I get this error on the prometheus pod when testing :
```
MountVolume.SetUp failed for volume "prometheus-config-volume" : configmap "prometheus-config" not found
```
And the pod never starts
Screen Shot 2019-12-20 at 12 36 48 PM

**Comment by nitisht** [comment]
You need to use the updated m3-deployment.yaml file from the example @cesnietor

**Comment by cesnietor** [comment]
> You need to use the updated m3-deployment.yaml file from the example @cesnietor

oh I see lol, testing again.

**Comment by cesnietor** [comment]
LGTM tested and it shows the tenants on the `Service Discovery` -> `minio-metrics`
Screen Shot 2019-12-20 at 1 00 23 PM

**Review by cesnietor — DISMISSED** [review]

**Comment by harshavardhana** [comment]
should we take this @dvaldivia @cesnietor ?

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**